lib/transform.js

```diff
@@ -64,7 +64,7 @@ TreeTransformer.prototype = new TreeWalker;
                     x = this;
                     descend(x, tw);
                 } else {
-                    tw.stack[tw.stack.length - 1] = x = this.clone();
+                    tw.stack[tw.stack.length - 1] = x = this;
                     descend(x, tw);
```

Comment:
Yeah, noticed that the tests fail when I undo that.  I forgot why I was cloning there, for the time being I'm alright with the patch.

Thanks again for this and your past contributions!  If you are interested to have full commit rights, please ping me in private ([my blog](http://lisperator.net/) tells you how to find me).